### PR TITLE
add "set up direct deposit" state

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Updates the view users see when they are eligible to set up Direct Deposit but have not yet done so.

## Testing done
Local + unit tests. Refactored the existing unit tests and added new tests to make sure that the "not set up" state renders correctly and moving from the "not set up" state to the "edit" state and back works.

## Screenshots
direct deposit not set up

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx [hunk 1]
```diff
@@ -138,15 +138,15 @@ export const DirectDepositContent = ({
 
   // When direct deposit is not set up, we will show
   const notSetUpContent = (
-    <>
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx:141]: no idea why this button was wrapped in a fragment before.

File: src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx [hunk 2]
```diff
@@ -138,15 +138,15 @@ export const DirectDepositContent = ({
...
-        onClick={() => {
-          toggleEditState();
-        }}
-      >
-        Set up direct deposit
-      </button>
-    </>
+    <button
+      className="va-button-link"
+      ref={editBankInfoButton}
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx:143]: without adding this ref the app would crash when returning from the edit view back to the not-set-up state

File: src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx [hunk 3]
```diff
@@ -195,22 +195,27 @@ export const DirectDepositContent = ({
...
+  const directDepositData = () => {
+    const data = [
+      // top row of the table can show multiple states so we set its value with
+      // the getBankInfo() helper
+      {
+        title: 'Account',
+        value: getBankInfo(),
+      },
+    ];
+    if (isDirectDepositSetUp) {
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/components/direct-deposit/DirectDepositContent.jsx:207]: only show the Payment History link if DD is set up

**Review by sanlouise — APPROVED** [review]
Looks great, and thank you for adding such exhaustive testing coverage. Works as expected locally. 👍

**Pull request merged**